src/agent/Core/Controller/CheckoutSession.cpp

```diff
@@ -353,9 +353,13 @@ Controller::writeRequestQueueFullExceptionErrorResponse(Client *client, Request
 		" due to: " << e->what());
 
 	endRequestWithSimpleResponse(&client, &req,
+		"<html><head>"
```

Comment:
What do you think of adding an id with a machine-readable code that downstream applications can detect when applying translations, etc?

```suggestion
		"<html id="error-503-queue-full"><head>"
```

This would be helpful with cases like https://josm.openstreetmap.de/ticket/24346 (if I understand their issue correctly).

Reply:
It is probably better for OSM to check the http status and replace the whole page, if they'd like to have more control of the contents.

Reply:
Are there other reasons why a 503 could be triggered, apart from a full queue? Asking as it stands for "Service Unavailable", which technically could happen for other reasons.

Reply:
There are, but you can control which status is used for queue overflow with passenger_request_queue_overflow_status_code

Reply:
But that wouldn't solve that problem, right? It would move the problem from one crowded HTTP status to another. So for example: here we have several potential meanings for a status 503: it could be a full queue, it could be OOM (no idea, just speculating). If we set `passenger_request_queue_overflow_status_code=500`, then there would again be several reasons why a 500 is happening. A service downstream wouldn't be able to tell what specific reason triggered the 500, out of several possibilities.

However your feedback does make me realise that, if this `id` were to be added, it shouldn't reference 503 directly because it could be a different one.

But anyway, I also realise that I'm bikeshedding in an area I know absolutely nothing about :flushed: So I'm going to stop here. Thank you for the review in the first place! :heart: